main.rb

```diff
@@ -0,0 +1,95 @@
+module Enumerable
+
+    def my_each
+    unless !block_given?
+      for item in self
+        yield(item)
+      end
+    end
+      return self
+    end
+  
+    def my_each_with_index
+    unless !block_given?
+      for item_index in (0...self.size)
+        yield(self[item_index], item_index)
+      end
+    end
+      return self
+    end
+  
+    def my_select
+    unless !block_given?
+      new_items = []
+      my_each do |item|
+        new_items.push(item) if yield(item)
+      end
+      return new_items
+    end
+      return self
+    end
+  
+    def my_all?
+    return true if !block_given?
+    items_exist = true
+      my_each do |item|
+            items_exist = false if !yield(item)
+      end
+    items_exist
+    end
+  
+    def my_count
+      return self.size if !block_given?
+      count = 0
+      my_select do |item|
+          count+=1 if yield(item)
+      end
+      count
+    end
```

Comment:
`my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- counts the number of items in enum that are equal to item if an argument is given
```ruby
[1, 1, 2 ,3].my_count(1)  #should return 2
```